### PR TITLE
Refactor Physical Body Handling

### LINKED ISSUE #64: End Astral Travel when taking drowning damage

### LINKED ISSUE #90: Extra hearts are removed after merging with your body
When you have extra hearts with mods like SoL: Carrot Edition, you lose the bonus hearts when using Astral Travel

### LINKED ISSUE #91: Creative Player can still die while in Astral Form
`Minecraft v1.15.2`
`Forge v31.2.31`
`Astral v1.4.15`

If the player's body dies or takes damage, the player will die or take damage in creative mode.

### PR DESCRIPTION
**Pull request by Alan19**
- Body now dies when player is not logged in
- Player cap keeps track of body info right before it dies, world cap keeps track of body NBT to fill in the rest of the info
- Block damage to bodies in creative mode
- Drowning damage ends Astral Travel

### CODE REVIEW

File: src/main/java/com/alan19/astral/api/bodylink/BodyLink.java [hunk 1]
```diff
@@ -38,104 +41,58 @@
...
-    public CompoundNBT serializeNBT() {
-        CompoundNBT mapNBT = new CompoundNBT();
-        for (Map.Entry<UUID, BodyInfo> bodyInfoEntry : bodyInfoMap.entrySet()) {
-            mapNBT.put(bodyInfoEntry.getKey().toString(), bodyInfoEntry.getValue().serializeNBT());
-        }
-        return mapNBT;
+    public Optional<PhysicalBodyEntity> getBody(ServerWorld world) {
+        final ServerWorld bodyWorld = world.getServer().getWorld(bodyInfo.getDimensionType());
+        final Chunk chunk = bodyWorld.getChunkAt(bodyInfo.getPos());
+        chunk.setLoaded(true);
```
Comment (SkySom) [inline src/main/java/com/alan19/astral/api/bodylink/BodyLink.java]: This isn't great. You shouldn't be loading chunks. As if they don't deserve to be loaded, MC will unload them, which can lead to lots of load/unloading

File: src/main/java/com/alan19/astral/events/astraltravel/StartAndEndHandling.java [hunk 2]
```diff
@@ -140,11 +157,16 @@ public static void spawnPhysicalBody(PlayerEntity playerEntity) {
...
                 psychicInventory.ifPresent(iPsychicInventory -> iPsychicInventory.getInventoryOfPlayer(playerEntity.getUniqueID()).setInventoryType(InventoryType.ASTRAL, playerEntity.inventory));
             }
-            physicalBodyEntity.setHealth(playerEntity.getHealth());
-            physicalBodyEntity.setHungerLevel(playerEntity.getFoodStats().getFoodLevel());
         }
     }
 
+    public static void spawnPhysicalBody(PlayerEntity playerEntity, BodyInfo bodyInfo, CompoundNBT nbt) {
+        nbt.putFloat("Health", bodyInfo.getHealth());
+        final ListNBT pos = nbt.getList("Pos", NBT.TAG_DOUBLE);
```
Comment (SkySom) [inline src/main/java/com/alan19/astral/events/astraltravel/StartAndEndHandling.java:165]: storing a pos as a long is usually what most do I believe.

**Review by SkySom — APPROVED** [review]

File: src/main/java/com/alan19/astral/events/astraltravel/StartAndEndHandling.java (hunk 2 above)
Comment (Alan19) [reply src/main/java/com/alan19/astral/events/astraltravel/StartAndEndHandling.java:165]: Entity writes the pos as `compound.put("Pos", this.newDoubleNBTList(this.getPosX(), this.getPosY(), this.getPosZ()));`, so it should be fine

File: src/main/java/com/alan19/astral/api/bodylink/BodyLink.java (hunk 1 above)
Comment (Alan19) [reply src/main/java/com/alan19/astral/api/bodylink/BodyLink.java]: Got rid of the loading code

**Pull request merged**